### PR TITLE
Fix/audit external

### PR DESCRIPTION
**Pull request by EdoWahdana**

### CODE REVIEW

File: paras-marketplace-contract/src/lib.rs [hunk 1]
```diff
@@ -433,7 +433,7 @@ impl Contract {
             "Paras: NEAR support only"
         );
 
-        assert!(market_data.is_auction.is_none(), "Paras: the NFT is on auction");
+        assert_eq!(market_data.is_auction.unwrap(), false, "Paras: the NFT is on auction");
```
Comment (emarai) [inline paras-marketplace-contract/src/lib.rs:436]: please check : 

if is_auction.is_none() -> allow buy
if is_auction.is_some() && is_auction.unwrap() == false -> allow buy

use if let Some() .. for cleaner code

File: paras-marketplace-contract/src/lib.rs [hunk 2]
```diff
@@ -1651,19 +1651,15 @@ impl Contract {
         let bidder_id = env::predecessor_account_id();
         let current_time = env::block_timestamp();
 
-        if market_data.started_at.is_some() {
-            assert!(
-                current_time >= market_data.started_at.unwrap(),
-                "Paras: Sale has not started yet"
-            );
-        }
+        assert!(
```
Comment (emarai) [inline paras-marketplace-contract/src/lib.rs:1654]: use if let Some() for cleaner code

File: paras-marketplace-contract/src/lib.rs [hunk 3]
```diff
@@ -1969,26 +1965,26 @@ impl Contract {
 
         let current_time: u64 = env::block_timestamp();
 
-        if started_at.is_some() {
-            assert!(started_at.unwrap().0 >= current_time);
+        if let Some(is_auction) = is_auction {
+            if started_at.is_some() {
```
Comment (emarai) [inline paras-marketplace-contract/src/lib.rs:1969]: use if let Some() also here

File: paras-marketplace-contract/src/lib.rs [hunk 4]
```diff
@@ -1651,19 +1651,15 @@ impl Contract {
...
         let current_time = env::block_timestamp();
 
-        if market_data.started_at.is_some() {
-            assert!(
-                current_time >= market_data.started_at.unwrap(),
-                "Paras: Sale has not started yet"
-            );
-        }
+        assert!(
+            current_time >= market_data.started_at.unwrap(),
```
Comment (joundy) [inline paras-marketplace-contract/src/lib.rs:1655]: it's possible to check the started_at and ended_at values only when is_auction true?

File: paras-marketplace-contract/src/lib.rs (hunk 1 above)
Comment (joundy) [inline paras-marketplace-contract/src/lib.rs:436]: btw ini bisa di define dulu is_auctionya biar pas handle di bawahnya lebih gampang
let is_auction = market_data.is_auction.unwrap_or(false);

**Pull request merged**